src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
This fragment of code looks really complex to read.
Better to use `Array.prototype.reduce` and store result  of `people.find` into the variable
```
  const res = children.reduce((sum, ch) => {
      const motherBorn = people.find((m) => m.name === ch.mother).born;
      sum += ch.born - motherBorn
      return sum;
  }, 0);
```